components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/function/ResourcesToValidatedOrigins.java

```diff
@@ -0,0 +1,40 @@
+package org.wso2.carbon.identity.cors.mgt.core.internal.function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resources;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceClientException;
+import org.wso2.carbon.identity.cors.mgt.core.model.ValidatedOrigin;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Converts a ConfigurationManagement Resources to a list of ValidatedOrigins.
+ */
+public class ResourcesToValidatedOrigins implements Function<Resources, List<ValidatedOrigin>> {
+
+    private static final Log log = LogFactory.getLog(ResourcesToValidatedOrigins.class);
+
+    @Override
+    public List<ValidatedOrigin> apply(Resources resources) {
+
+        ArrayList<ValidatedOrigin> validatedOrigins = new ArrayList<>();
+        for (Resource resource : resources.getResources()) {
+            try {
+                ValidatedOrigin validatedOrigin = new ValidatedOrigin(resource.getResourceName());
+                validatedOrigins.add(validatedOrigin);
+            } catch (CORSManagementServiceClientException e) {
+                if (log.isDebugEnabled()) {
+                    // The program should never reach here as all the origins in the configuration management store
+                    // are properly validated.
+                    log.debug(e);
```

Comment:
let's add an informative error message as well

Reply:
The error already has a message.